runtime/util/gcloadinfo.c

```diff
@@ -0,0 +1,45 @@
+/*******************************************************************************
+ * Copyright (c) 2020, 2020 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include "gcloadinfo.h"
+
+/**
+ * Retrieves the load info for the appropriate GC DLL based on reference mode.
+ *
+ * @param vm The Java VM
+ * @returns J9VMDllLoadInfo for the GC DLL selected
+ */
+J9VMDllLoadInfo *
+getGCDllLoadInfo(J9JavaVM *vm)
+{
+	J9VMDllLoadInfo *loadInfo = NULL;
+	const char *gcDLLName = J9_GC_DLL_NAME;
+
+#if defined(OMR_MIXED_REFERENCES_MODE_STATIC)
+	if (!J9JAVAVM_COMPRESS_OBJECT_REFERENCES(vm)) {
+		gcDLLName = J9_GC_FULL_DLL_NAME;
+	}
+#endif /* defined(OMR_MIXED_REFERENCES_MODE_STATIC) */
+	loadInfo = FIND_DLL_TABLE_ENTRY(gcDLLName);
```

Comment:
@gacholio I think there's something wrong here. This should be the pattern instead?
```
#if defined(compressed_pointers) && defined(full_pointers) // in mixed mode
#if defined(mixed_mode_static)
	if (-Xnocompressedrefs was specified)
                gcDLLName = GC_FULL;
#else // dynamic mode
	if (!compressed_macro_check()) {
		gcDLLName = GC_FULL;
	}
#endif /* defined(mixed_mode_static) */
#endif /* defined(compressed_pointers) && defined(full_pointers) */
```

Reply:
In dynamic mode, there will only be a single DLL, much like the existing builds. The existing code seems correct to me.

Reply:
Ok good to confirm that. So, I think there's something up with `OMR_MIXED_REFERENCES_MODE_STATIC`. It seems to not be defined in these files...but it is enabled in the CMakeCache.txt `OMR_MIXED_REFERENCES_MODE_STATIC:INTERNAL=ON`. Building the `*_full` versions of the GC libs is also ifdeffed on `OMR_MIXED_REFERENCES_MODE_STATIC`, and those are all getting built.

I'm not really sure why this is occurring, but I'm assuming some CMake config is a bit off. Looking into this more.

Reply:
I was missing a change to make `OMR_MIXED_REFERENCES_MODE_STATIC` accessible in the omr config.